src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -128,7 +129,9 @@
   [database]
   {:pre [(or nil? (instance? (type Database) database))]}
   (if (some? database)
-    (hash (:details database))
+    (hash [(:details database)
+           (driver/report-timezone)
+           (public-settings/start-of-week)])
```

Comment:
Report timezone and start of week are now included in the DB details hash which is used to determine whether the current connection pool for a database is still valid or whether we need to drop it and create a new one. Previously we only hashed the Database `:details` column, but that didn't accurately include stuff that goes in the JDBC spec (e.g. the `WEEK_START` option for Snowflake, which is not based on `:details`). 

Probably the actual correct thing to do here is hash the JDBC connection spec instead of `:details`. I'm going to look into that, but will probably do it as a separate PR either way